src/GeekLearning.Authorizations.EntityFrameworkCore/AuthorizationsManager.cs

```diff
@@ -317,13 +333,31 @@ await this.context.Set<Data.ScopeHierarchy>()
                     this.context.Set<Data.Membership>().Remove(memberShip);
                 }
 
-                await this.UnaffectRolesFromGroupAsync(groupName);
+                await this.UnaffectRolesFromGroupAsync(group.Name);
 
                 this.context.Set<Data.Group>().Remove(group);
                 this.context.Set<Data.Principal>().Remove(group.Principal);
             }
         }
 
+        public async Task AddPrincipalToGroupAsync(Guid principalId, Guid groupId)
+        {
+            var membership = await this.GetEntityAsync<Data.Membership>(m => m.PrincipalId == principalId && m.Group != null && m.GroupId == groupId);
+            if (membership == null)
+            {
+                var group = await this.GetEntityAsync<Data.Group>(g => g.Id == groupId);
+                this.context.Set<Data.Membership>().Add(new Data.Membership
+                {
+                    PrincipalId = principalId,
+                    GroupId = groupId,
+                    CreationBy = this.principalIdProvider.PrincipalId,
+                    ModificationBy = this.principalIdProvider.PrincipalId
+                });
+
+                this.QueueEvent(new AddPrincipalToGroup(principalId, group.Name));
```

Comment:
Should be based on Group ID not name. Group name is not invariant and can be edited.